### PR TITLE
Localpacking2

### PR DESCRIPTION
**Pull request by svemeyer**

### CODE REVIEW

File: packer/src/verify_container.py [hunk 1]
```diff
@@ -0,0 +1,408 @@
...
+    md5_value = md5()
+    with open(filepath, "rb") as file:
+        for chunk in iter(lambda: file.read(4096), b""):
+            md5_value.update(chunk)
+    return base64.b64encode(md5_value.digest()).decode()
+
+
+def _adler32(filepath):  # Does this work for large files?
+    with open(filepath, "rb") as file:
+        adler32_value = adler32(file.read())
```
Comment (kofemann) [inline packer/src/verify_container.py]: you have to read in chunks here as well.
see example at https://gist.github.com/kofemann/2303046#file-adler32-py

File: packer/src/verify_container.py [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+            db = client[mongo_db]
+            with db.archives.find() as db_archives:
+                skip = False
+                logger.info(f"Found {db.archives.count_documents({})} new archives")
+                for archive in db_archives:
+                    if not running:
+                        logger.info("Exiting")
+                        sys.exit(0)
+                    url = f"{webdav_door}/{archive['dest_path']}/{os.path.basename(archive['path'])}"
+                    # Open ZIP-File and get filelist
```
Comment (kofemann) [inline packer/src/verify_container.py:189]: I guess this block is an attempt to re-start interrupted container.  May be move it into a dedicated function?

File: packer/src/verify_container.py [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+                                           f"Creating new entry to failures collection")
+                            db.failures.insert_one({'archivePath': archive['path'], 'pnfsid': pnfsid})
+                        elif pnfsid in db_pnfsidlist:
+                            logger.warning(f"File {pnfsid} is listed in MongoDB to be in archive {archive['path']}, "
+                                           f"but isn't there. Resetting file for packing into new archive")
+                            db_file = db.files.find_one({"pnfsid": pnfsid})
+                            db_file['state'] = "new"
+                            db.files.replace_one({"pnfsid": pnfsid}, db_file)
+
+                    # Upload zip-file to dCache
```
Comment (kofemann) [inline packer/src/verify_container.py:233]: This section as well should go into a dedicated function

**Review by kofemann — CHANGES_REQUESTED** [review]
A general comment: please split the packing loop into smaller functions.

File: packer/src/verify_container.py (hunk 2 above)
Comment (svemeyer) [reply packer/src/verify_container.py:189]: "if running" is used for KeyInterrupt. If you stop the script with SIGINT, running is set to False, so the script finishes the archive and stops before it takes the next one. url is the url for all webdav requests

**Pull request merged**